src/qt/askpassphrasedialog.cpp

```diff
@@ -74,6 +74,12 @@ AskPassphraseDialog::AskPassphraseDialog(Mode _mode, QWidget *parent) :
     connect(ui->passEdit1, &QLineEdit::textChanged, this, &AskPassphraseDialog::textChanged);
     connect(ui->passEdit2, &QLineEdit::textChanged, this, &AskPassphraseDialog::textChanged);
     connect(ui->passEdit3, &QLineEdit::textChanged, this, &AskPassphraseDialog::textChanged);
+
+    // always use modal userflow blocking
+    setModal(true);
+
+    // auto-delete instance after close event
+    setAttribute(Qt::WA_DeleteOnClose);
```

Comment:
I don't think this should be set in the constructor. It is not the dialog's responsibility to know about its own object lifetime. So I think it's more clear to do so at the call site.

Reply:
This was suggested by me in https://github.com/bitcoin/bitcoin/pull/15313#discussion_r253711302 and now I agree with you. @jonasschnelli sorry ☺️

Reply:
Yes. I agree with @laanwj. Fixed.